packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/distant-table/distant-table.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { EntityManager, Repository } from 'typeorm';
+import { v4 } from 'uuid';
+
+import {
+  RemoteServerEntity,
+  RemoteServerType,
+} from 'src/engine/metadata-modules/remote-server/remote-server.entity';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+import { DistantTableColumn } from 'src/engine/metadata-modules/remote-server/remote-table/distant-table/types/distant-table-column';
+import { DistantTables } from 'src/engine/metadata-modules/remote-server/remote-table/distant-table/types/distant-table';
+
+@Injectable()
+export class DistantTableService {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    @InjectRepository(RemoteServerEntity, 'metadata')
+    private readonly remoteServerRepository: Repository<
+      RemoteServerEntity<RemoteServerType>
+    >,
+  ) {}
+
+  public async fetchDistantTableColumns(
+    remoteServer: RemoteServerEntity<RemoteServerType>,
+    tableName: string,
+  ): Promise<DistantTableColumn[]> {
+    if (!remoteServer.availableTables) {
+      throw new Error('Remote server available tables are not defined');
+    }
+
+    return remoteServer.availableTables[tableName];
+  }
+
+  public async fetchDistantTableNames(
+    remoteServer: RemoteServerEntity<RemoteServerType>,
+    workspaceId: string,
+  ): Promise<string[]> {
+    if (!remoteServer.schema) {
```

Comment:
I don't see where we ask the schema name from the UI. Is it planned for an upcoming UI? Currently it's failing with this check

Reply:
Also, I wouldn't throw an "Error" here, this does not seem to be internal issue but more tied to the user input and we don't want to capture such exceptions to Sentry for example.
Can you throw a BadRequestException? Ideally in the future we will have specific exceptions internal to twenty and not tied to REST/Graphql implementations but in the meantime let's use that!

Reply:
>Ideally in the future we will have specific exceptions internal to twenty and not tied to REST/Graphql implementations but in the meantime let's use that!

cc @charlesBochet 😄

Reply:
Waiting on @ijreilly's work to add the schema as input!

Reply:
Finally added it in this PR